pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -289,6 +305,66 @@ func (a *Actuator) Delete(ctx context.Context, cluster *clusterv1.Cluster, machi
 	return nil
 }
 
+func (a *Actuator) cordon(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine, scope *actuators.MachineScope) (error, error) {
+	a.log.Info("Inside cordon()")
+
+	node, err := a.getNodeObject(ctx, cluster, machine, scope)
+	if err != nil {
+		a.log.Info("ERROR: ", err.Error())
+		return nil, err
+	}
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		a.log.Info("ERROR: ", err.Error())
+		return nil, err
+	}
+
+	clientSet, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		a.log.Info("ERROR: ", err.Error())
+		return nil, err
+	}
+
+	helper := drain.NewCordonHelper(node)
+	if helper.UpdateIfRequired(true) {
+		a.log.Info("CordonHelper: ", node.Name)
+		return helper.PatchOrReplace(clientSet)
+	}
+
+	return nil, nil
+}
+
+func (a *Actuator) getNodeObject(ctx context.Context, cluster *clusterv1.Cluster, machine *clusterv1.Machine, scope *actuators.MachineScope) (*apicorev1.Node, error) {
+	coreClient, err := a.coreV1Client(scope.Cluster)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to retrieve corev1 client for cluster %q", scope.Cluster.Name)
+	}
+
+	listOpt := metav1.ListOptions{}
+
+	for {
+		nodeList, err := coreClient.Nodes().List(listOpt)
```

Comment:
Similar to the cordon comment, this will only work for nodes that exist in the same cluster as the machine objects (self-managed clusters).

Reply:
It's been already fixed.